web/packages/teleterm/src/deepLinks.ts

```diff
@@ -104,20 +84,51 @@ export function parseDeepLink(rawUrl: string): DeepLinkParseResult {
     };
   }
 
-  if (whatwgURL.pathname !== '/connect_my_computer') {
-    return { status: 'error', reason: 'unsupported-uri' };
-  }
-
-  const { host, hostname, port, username, pathname } = whatwgURL;
-  const url: DeepURL = {
+  // TODO (avatus): remove when authenticate web device case is implemented
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { host, hostname, port, username, pathname, searchParams } = whatwgURL;
+  const baseUrl = {
     host,
     hostname,
     port,
     // whatwg-url percent-encodes usernames. We decode them here so that the rest of the app doesn't
     // have to do this. https://url.spec.whatwg.org/#set-the-username
     username: decodeURIComponent(username),
-    pathname,
   };
 
-  return { status: 'success', url };
+  switch (pathname as Path) {
+    case '/connect_my_computer': {
+      const url: ConnectMyComputerDeepURL = {
+        ...baseUrl,
+        pathname: '/connect_my_computer',
```

Comment:
Quite unfortunate that we have to repeat the pathname, it looks like the type assertion in `switch` makes its type to be a string :/

I tried to fix this, but if I remove  `as Path` then the cases are no longer typed.